sql/database.go

```diff
@@ -292,16 +330,194 @@ func Open(uri string, opts ...Opt) (*sqliteDatabase, error) {
 	return db, nil
 }
 
+func ensureDBSchemaUpToDate(logger *zap.Logger, db *sqliteDatabase, config *conf) (*sqliteDatabase, error) {
+	before, after, err := config.schema.CheckDBVersion(logger, db)
+	switch {
```

Comment:
this `switch` has been repurposed to do general flow-control rather than checking for a few correlated values. This is probably some of the properties of the golang switch that I am less of a fan (in other languages you can only usually use a switch over a single value and do pattern matching over it, while in go you can do whatever...)

Reply:
Well, maybe that's a matter of taste... The direct alternative is many-branched `if ... { ... } else if ... { ... } else if ... { ... }` and I'm not sure some extra punctuation really makes it much easier to read as opposed to all the possibilities listed in `case` clauses. Granted, it's usually better to avoid situations where we need so many branches in the first place, but in some cases that's not very easy to achieve

Reply:
I actually also quite like this style of testing for a lot of different (mutually exclusive) conditions. I feel like it is easier to see that only one (or none) of the `cases` is expected to be true.